src/search/landmarks/landmark_factory_h_m.cc

```diff
@@ -588,78 +589,25 @@ void LandmarkFactoryHM::initialize(const TaskProxy &task_proxy) {
     build_pm_ops(task_proxy);
 }
 
-void LandmarkFactoryHM::generate(const TaskProxy &task_proxy) {
-    if (only_causal_landmarks) {
-        Exploration exploration(task_proxy);
-        discard_noncausal_landmarks(task_proxy, exploration);
-    }
-    if (!disjunctive_landmarks)
-        discard_disjunctive_landmarks();
+void LandmarkFactoryHM::postprocess(const TaskProxy &task_proxy) {
     if (!conjunctive_landmarks)
         discard_conjunctive_landmarks();
     lm_graph->set_landmark_ids();
 
-    if (no_orders)
+    if (!use_orders)
```

Comment:
This change here confirms my thoughts about `use_orders` and `reasonable_orders` being on a similar level. They both denote changes on the landmark graph after its generation. More so, this is also the case for the noncausal, disjunctive, and conjunctive landmarks options. To treat these things similarly, it would make sense to keep reasonable orders as an option I think.

Reply:
Hmm. Reading your two comments regrading this change, I thought you were going to suggest to computing orders in a separate factory, too. 

But maybe I don't understand enough anyway: this part of the code looks like orders are generated anyway and only discarded if the option is not set. So using a order-factory after (or rather "wrapping") the landmark factory wouldn't work in the same way, would it? But I guess it could be fine to just drop landmarks from a previously computed graph...

Reply:
Ok, no that I have seen how large the new factory for computing reasonable orders is, I think it makes sense that it is a separate class. Also, since we are working on making landmark graphs acyclic in a separate class, too, I would stick with this being its own factory. The question then is if use_orders should be its own factory, too. As said above, I'm probably not familiar enough with the details, but if we tend towards having "postprocessing" factories in the future, then use_orders (or better: discard orders) could indeed be one of them.

Speaking of `use_orders`: I would consider to change it "back", i.e., to `discard_orders` because this better describes what the option actually does and it goes along with `discard_conjunctive_landmarks` and `make_acyclic` because all of these remove landmarks rather then computing new ones or forbidding certain types during construction. (Unless I understood this wrong, of course.)

Reply:
A final thought on the factory vs option thing: for the new/planned factories which drop landmarks, we could consider not making them landmark graph factories but "postprocessing factories" or similar to distinguish them from the "generating" factories.

Reply:
I think that the option `conjunctive_landmarks`is also not ideal. Does it mean we compute these? Keep these? I agree that either `keep/compute_conjunctive_landmarks` or `discard_conjunctive_landmarks` is long, but I would still go with the longer variants. Or abbreviate them to `keep/compute_conj_lms` as the code sometimes does.

I'm in favor of that variant where we don't have to negate the option in the code. So if we only ask `if (!keep_...)`, then discard, I think the option should be named the opposite way.

Reply:
> I'm in favor of that variant where we don't have to negate the option in the code. So if we only ask if (!keep_...), then
> discard, I think the option should be named the opposite way.

Trying to use positive rather than negative conditions is a good principle, but I think we should care about the user interface (= the command-line options) more than we should care about the implementation details.

Generating/keeping conjunctive landmarks is expressed more positively than discarding them, so I think that's the more obvious of the two choices. Also, keeping the "direction" of the flag in the user interface the way it always has been is preferable unless there is a strong reason to change it.